### ISSUE TITLE
dune build crashes badly with weird coq stanza

### ISSUE THREAD

**Issue by Alizter** [ocaml/dune#5530]
## Expected Behavior

Running dune build doesn't crash.

## Actual Behavior

Running dune build throws an internal error:

```ocaml
Internal error, please report upstream including the contents of _build/log.
Description:
  ("Map.find_exn: failed to find key",
  { key = [ "HoTT"; "Contrib" ]; keys = [] })
Raised at Stdune__code_error.raise in file "otherlibs/stdune/code_error.ml",
  line 11, characters 30-62
Called from Dune_rules__coq_sources.directories in file
  "src/dune_rules/coq_sources.ml" (inlined), line 40, characters 26-70
Called from Dune_rules__coq_rules.setup_rules in file
  "src/dune_rules/coq_rules.ml", line 429, characters 22-68
Called from Fiber.Scheduler.exec in file "src/fiber/fiber.ml", line 854,
  characters 10-13
-> required by ("gen-rules", "default")
-> required by ("load-dir", In_build_dir "default")
-> required by ("build-alias", { dir = "default"; name = "default" })
-> required by ("toplevel", ())

I must not crash.  Uncertainty is the mind-killer. Exceptions are the
little-death that brings total obliteration.  I will fully express my cases. 
Execution will pass over me and through me.  And when it has gone past, I
will unwind the stack along its path.  Where the cases are handled there will
be nothing.  Only I will remain.
```

## Reproduction

Clone this branch:
https://github.com/Alizter/HoTT/tree/dune-bug-1

Run dune build in HoTT.

The problem is likely in `HoTT/contrib/dune` where I wrote a bit of a nonsense stanza:
```ocaml
(coq.theory
 (name HoTT.Contrib) ; <== Likely the issue
 (synopsis "The HoTT Coq library")
 (libraries coq-core.plugins.ltac HoTT)
 (flags -noinit -indices-matter -color on))
```

## Specifications

- Version of `dune` (output of `dune --version`): 3.0.3
- Version of `ocaml` (output of `ocamlc --version`) 4.13.1
- Version of `coq` (output of `coqc --version`)
```
The Coq Proof Assistant, version 8.15.0
compiled with OCaml 4.13.1
```
- Operating system (distribution and version): 
```
N o LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.4 LTS
Release:        20.04
Codename:       focal
````

## Additional information 

- Link to gist with verbose output (run `dune` with the `--verbose` flag):

<details>
<summary>Details</summary>

```
Shared cache: enabled
Workspace root: /home/ali/HoTT
Running[0]: /mnt/sdd1/.opam/coq.8.15.0/bin/ocamlc.opt -config > /tmp/dune_23ed2d_output
Dune context:                   
 { name = "default"
 ; kind = "default"
 ; profile = Dev
 ; merlin = true
 ; for_host = None
 ; fdo_target_exe = None
 ; build_dir = "default"
 ; toplevel_path = Some External "/mnt/sdd1/.opam/coq.8.15.0/lib/toplevel"
 ; ocaml_bin = External "/mnt/sdd1/.opam/coq.8.15.0/bin"
 ; ocaml = Ok External "/mnt/sdd1/.opam/coq.8.15.0/bin/ocaml"
 ; ocamlc = External "/mnt/sdd1/.opam/coq.8.15.0/bin/ocamlc.opt"
 ; ocamlopt = Ok External "/mnt/sdd1/.opam/coq.8.15.0/bin/ocamlopt.opt"
 ; ocamldep = Ok External "/mnt/sdd1/.opam/coq.8.15.0/bin/ocamldep.opt"
 ; ocamlmklib = Ok External "/mnt/sdd1/.opam/coq.8.15.0/bin/ocamlmklib.opt"
 ; env =
     map
       { "DUNE_OCAML_HARDCODED" : "/mnt/sdd1/.opam/coq.8.15.0/lib"
       ; "DUNE_OCAML_STDLIB" : "/mnt/sdd1/.opam/coq.8.15.0/lib/ocaml"
       ; "DUNE_SOURCEROOT" : "/home/ali/HoTT"
       ; "INSIDE_DUNE" : "/home/ali/HoTT/_build/default"
       ; "OCAMLFIND_IGNORE_DUPS_IN" :
           "/home/ali/HoTT/_build/install/default/lib"
       ; "OCAMLPATH" : "/home/ali/HoTT/_build/install/default/lib"
       ; "OCAMLTOP_INCLUDE_PATH" :
           "/home/ali/HoTT/_build/install/default/lib/toplevel"
       ; "OCAML_COLOR" : "always"
       ; "OPAMCOLOR" : "always"
       }
 ; findlib_path = [ External "/mnt/sdd1/.opam/coq.8.15.0/lib" ]
 ; arch_sixtyfour = true
 ; natdynlink_supported = true
 ; supports_shared_libraries = true
 ; ocaml_config =
     { version = "4.13.1"
     ; standard_library_default = "/mnt/sdd1/.opam/coq.8.15.0/lib/ocaml"
     ; standard_library = "/mnt/sdd1/.opam/coq.8.15.0/lib/ocaml"
     ; standard_runtime = "the_standard_runtime_variable_was_deleted"
     ; ccomp_type = "cc"
     ; c_compiler = "gcc"
     ; ocamlc_cflags =
         [ "-O2"; "-fno-strict-aliasing"; "-fwrapv"; "-pthread"; "-fPIC" ]
     ; ocamlc_cppflags = [ "-D_FILE_OFFSET_BITS=64" ]
     ; ocamlopt_cflags =
         [ "-O2"; "-fno-strict-aliasing"; "-fwrapv"; "-pthread"; "-fPIC" ]
     ; ocamlopt_cppflags = [ "-D_FILE_OFFSET_BITS=64" ]
     ; bytecomp_c_compiler =
         [ "gcc"
         ; "-O2"
         ; "-fno-strict-aliasing"
         ; "-fwrapv"
         ; "-pthread"
         ; "-fPIC"
         ; "-D_FILE_OFFSET_BITS=64"
         ]
     ; bytecomp_c_libraries = [ "-lm"; "-ldl"; "-lpthread" ]
     ; native_c_compiler =
         [ "gcc"
         ; "-O2"
         ; "-fno-strict-aliasing"
         ; "-fwrapv"
         ; "-pthread"
         ; "-fPIC"
         ; "-D_FILE_OFFSET_BITS=64"
         ]
     ; native_c_libraries = [ "-lm"; "-ldl" ]
     ; native_pack_linker = [ "ld"; "-r"; "-o" ]
     ; cc_profile = []
     ; architecture = "amd64"
     ; model = "default"
     ; int_size = 63
     ; word_size = 64
     ; system = "linux"
     ; asm = [ "as" ]
     ; asm_cfi_supported = true
     ; with_frame_pointers = false
     ; ext_exe = ""
     ; ext_obj = ".o"
     ; ext_asm = ".s"
     ; ext_lib = ".a"
     ; ext_dll = ".so"
     ; os_type = "Unix"
     ; default_executable_name = "a.out"
     ; systhread_supported = true
     ; host = "x86_64-pc-linux-gnu"
     ; target = "x86_64-pc-linux-gnu"
     ; profiling = false
     ; flambda = false
     ; spacetime = false
     ; safe_string = true
     ; exec_magic_number = "Caml1999X030"
     ; cmi_magic_number = "Caml1999I030"
     ; cmo_magic_number = "Caml1999O030"
     ; cma_magic_number = "Caml1999A030"
     ; cmx_magic_number = "Caml1999Y030"
     ; cmxa_magic_number = "Caml1999Z030"
     ; ast_impl_magic_number = "Caml1999M030"
     ; ast_intf_magic_number = "Caml1999N030"
     ; cmxs_magic_number = "Caml1999D030"
     ; cmt_magic_number = "Caml1999T030"
     ; natdynlink_supported = true
     ; supports_shared_libraries = true
     ; windows_unicode = false
     }
 }
Actual targets:                 
- alias @@default
Internal error, please report upstream including the contents of _build/log.
Description:
  ("Map.find_exn: failed to find key",
  { key = [ "HoTT"; "Contrib" ]; keys = [] })
Raised at Stdune__code_error.raise in file "otherlibs/stdune/code_error.ml",
  line 11, characters 30-62
Called from Dune_rules__coq_sources.directories in file
  "src/dune_rules/coq_sources.ml" (inlined), line 40, characters 26-70
Called from Dune_rules__coq_rules.setup_rules in file
  "src/dune_rules/coq_rules.ml", line 429, characters 22-68
Called from Fiber.Scheduler.exec in file "src/fiber/fiber.ml", line 854,
  characters 10-13
-> required by ("gen-rules", "default")
-> required by ("load-dir", In_build_dir "default")
-> required by ("build-alias", { dir = "default"; name = "default" })
-> required by ("toplevel", ())

I must not crash.  Uncertainty is the mind-killer. Exceptions are the
little-death that brings total obliteration.  I will fully express my cases. 
Execution will pass over me and through me.  And when it has gone past, I
will unwind the stack along its path.  Where the cases are handled there will
be nothing.  Only I will remain.
```

</details>

**Comment by ejgallego** [comment]
Umm, that should not crash. @Alizter why do you think it is a nonsense stanza, because you are already using `(include_subdirs qualified)` in `HoTT/dune` ?

**Comment by Alizter** [comment]
It was nonsense, since I was in the middle of changing it and dune crashed badly which was surprising. I mean nonsense as in "the content doesn't really matter, dune shouldn't crash like this".

**Comment by ejgallego** [comment]
I mean, what should the correct `name` field be?

**Comment by Alizter** [comment]
Seems to be fixed by 1fd19d78e8430e9640ad2c8104c9dc6bbfffec40
